### PR TITLE
add channel to OCPdistributionInfo

### PR DESCRIPTION
**Pull request by zhiweiyin318**
for the requirement https://github.com/open-cluster-management/backlog/issues/11698#issuecomment-836687894

### CODE REVIEW

**Review by hanqiuzh — APPROVED** [review]
thanks!

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go [hunk 1]
```diff
@@ -331,6 +331,10 @@ func (r *ClusterInfoReconciler) getOCPDistributionInfo() (clusterv1beta1.OCPDist
 		return ocpDistributionInfo, client.IgnoreNotFound(err)
 	}
 
+	ocpDistributionInfo.Channel, _, err = unstructured.NestedString(obj.Object, "spec", "channel")
+	if err != nil {
+		klog.Errorf("failed to get OCP channel in clusterVersion: %v", err)
```
Comment (hanqiuzh) [inline pkg/klusterlet/clusterinfo/clusterinfo_controller.go:336]: wait, if the field is not exist what will happen? it is optional

Comment (zhiweiyin318) [reply pkg/klusterlet/clusterinfo/clusterinfo_controller.go:336]: the ocpDistributionInfo.Channel will be empty string, be omitempty in json

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go [hunk 2]
```diff
@@ -331,6 +331,10 @@ func (r *ClusterInfoReconciler) getOCPDistributionInfo() (clusterv1beta1.OCPDist
 		return ocpDistributionInfo, client.IgnoreNotFound(err)
 	}
 
+	ocpDistributionInfo.Channel, _, err = unstructured.NestedString(obj.Object, "spec", "channel")
```
Comment (qiujian16) [inline pkg/klusterlet/clusterinfo/clusterinfo_controller.go:334]: can we convert it to a typed api and read from typed API?

**Pull request merged**